ResearchKit/Common/ORKSkin.m

```diff
@@ -113,21 +113,48 @@ ORKScreenType ORKGetScreenTypeForBounds(CGRect bounds) {
     return screenType;
 }
 
-ORKScreenType ORKGetScreenTypeForWindow(UIWindow *window) {
+ORKScreenType ORKGetHorizontalScreenTypeForBounds(CGRect bounds) {
+    ORKScreenType screenType = ORKScreenTypeiPhone6;
+    CGFloat minimumDimension = MIN(bounds.size.width, bounds.size.height);
+    if (minimumDimension < ORKiPhone4ScreenSize.width + 1) {
+        screenType = ORKScreenTypeiPhone4;
+    } else if (minimumDimension < ORKiPhone5ScreenSize.width + 1) {
+        screenType = ORKScreenTypeiPhone5;
+    } else if (minimumDimension < ORKiPhone6ScreenSize.width + 1) {
+        screenType = ORKScreenTypeiPhone6;
+    } else if (minimumDimension < ORKiPhone6PlusScreenSize.width + 1) {
+        screenType = ORKScreenTypeiPhone6Plus;
+    } else {
+        screenType = ORKScreenTypeiPad;
+    }
+    return screenType;
+}
+
+ORKScreenType ORKGetVerticalScreenTypeForWindow(UIWindow *window) {
     if (!window) {
         window = [[[UIApplication sharedApplication] windows] firstObject];
```

Comment:
I figured that you use the window array method instead of `UIApplication`'s `keyWindow` or `UIApplication`'s `delegate`'s `window` because we may need the `window` before the `keyWindow` is set (e.g., if a view controller loads programmatically on the app delegate to be assigned as the root view controller).